cal/cal/static/js/components/category-details.js

```diff
@@ -63,61 +61,34 @@ function CategoriesDetailCtrl($scope, $http, QueryService){
   }.bind(this);
 
   this.showDaily = function() {
-    $http({
-      method: 'GET',
-      url: this.timeseriesDay + '.json',
-      params: {
-        timezone: query_timezone,
-      }
-    }).
-    then(function successCallback(response) {
+    QueryService.populateDay('Category day' + _this.categoryId, 'Category', _this.categoryId, []).
+    then(function populate(data) {
       _this.timeStep = "day";
-      _this.averageHours = Math.round(((_this.categoryHours / response.data[0].length) * 100)) / 100;
-      var eventData = QueryService.populateData(response.data, 'Category');
-      _this.ctrlGraphData = eventData[0];
-      _this.showGraph(eventData[1]);
+      _this.averageHours = Math.round(((_this.categoryHours / data[0][0].values.length) * 100)) / 100;
+      _this.ctrlGraphData = data[0];
```

Comment:
I changed it, however the last [0] has to stay there because data[0] has to be a list of lines in order for d3 to display them correctly. So [0] gets the first list. This won't ever be reordered in the backend so this is fine.